nav2-usability/scenario-code/deeplink-pathparam/router.dart

```diff
@@ -18,6 +18,40 @@ class Book {
   Book(this.title, this.author);
 }
 
+class AppState extends ChangeNotifier {
+  Book? _selectedBook;
+
+  Book? get selectedBook => _selectedBook;
+
+  set selectedBook(Book? book) {
```

Comment:
I am sorry if I missed this in the previous PR.

why not just expose the selectedId? it feels weird sometimes the app change selectedBook and sometimes uses setSelected(id).

Reply:
Right now the BooksListScreen's `onTapped` is a `ValueChanged<Book>`, so `_handleBookTapped` doesn't know the ID of the book that was tapped.

Alternatively, we could add an `ID field to each book.

Reply:
@chunhtai could you review the skipping stacks PR too? https://github.com/flutter/uxr/pull/22

Reply:
BookListScreen should know id of each book, in this case the index of the book in the list. It can passes the id into the ValueChanged.

Reply:
Are you saying that AppState shouldn't expose getters and setters for the selected book, only the selected book ID?

Reply:
I changed AppState to only store the ID, but I still need certain API methods to set the selected book: https://github.com/flutter/uxr/pull/23/commits/a66fa4cd138de04c31f0c3e0a8b8cf59667eb9c9

Reply:
I think it can just be books[id], or a global function since the books is outside of the app state.